nifi-docs/src/main/asciidoc/administration-guide.adoc

```diff
@@ -2569,21 +2625,29 @@ If `CreatorOnly` is specified, then only the user that created the data is allow
 In order to use the `CreatorOnly` option, NiFi must provide some form of authentication. See the <<zk_access_control>>
 section below for more information on how to configure authentication.
 
-If NiFi is configured to run in a standalone mode, the `cluster-provider` element need not be populated in the _state-management.xml_
-file and will actually be ignored if they are populated. However, the `local-provider` element must always be present and populated.
-Additionally, if NiFi is run in a cluster, each node must also have the `cluster-provider` element present and properly configured.
-Otherwise, NiFi will fail to startup.
+==== Kubernetes ConfigMap Cluster State Provider
 
-While there are not many properties that need to be configured for these providers, they were externalized into a separate _state-management.xml_
-file, rather than being configured via the _nifi.properties_ file, simply because different implementations may require different properties,
-and it is easier to maintain and understand the configuration in an XML-based file such as this, than to mix the properties of the Provider
-in with all of the other NiFi framework-specific properties.
+The Kubernetes ConfigMap State Provider supports shared cluster state when running in Kubernetes.
 
-It should be noted that if Processors and other components save state using the Clustered scope, the Local State Provider will be used
-if the instance is a standalone instance (not in a cluster) or is disconnected from the cluster. This also means that if a standalone instance
-is migrated to become a cluster, then that state will no longer be available, as the component will begin using the Clustered State Provider
-instead of the Local State Provider.
+The provider stores component state in
+link:https://kubernetes.io/docs/concepts/configuration/configmap/[Kubernetes ConfigMaps]
```

Comment:
I know this is indicated in the page that is linked here but we may want to also add the below note in our documentation as well:

> Note: A ConfigMap is not designed to hold large chunks of data. The data stored in a ConfigMap cannot exceed 1 MiB. If you need to store settings that are larger than this limit, you may want to consider mounting a volume or use a separate database or file service.

Thoughts?

Reply:
Thanks for the feedback @pvillard31! I agree that it is worth calling out the limit in NiFi documentation, so I added a note mentioning the 1 MB limit for ConfigMaps.